pkg/controller/nodes/dynamic/handler.go

```diff
@@ -441,42 +442,35 @@ func (d dynamicNodeTaskNodeHandler) progressDynamicWorkflow(ctx context.Context,
 		return handler.UnknownTransition, prevState, err
 	}
 
-	if state.HasFailed() {
+	if state.HasFailed() || state.HasTimedOut() {
+		// When the subworkflow either fails or times-out we need to handle failing
 		if dynamicWorkflow.GetOnFailureNode() != nil {
 			// TODO Once we migrate to closure node we need to handle subworkflow using the subworkflow handler
 			logger.Errorf(ctx, "We do not support failure nodes in dynamic workflow today")
 		}
 
-		return handler.DoTransition(handler.TransitionTypeEphemeral, handler.PhaseInfoRunning(nil)),
-			handler.DynamicNodeState{Phase: v1alpha1.DynamicNodePhaseFailing, Reason: state.Err.Error()},
+		// As we do not support Failure Node, we can just return failure in this case
```

Comment:
There was a good reason why we added this extra step in the internal statemachine, I wish I documented that :) let me try to remember, this node requires essentially two rounds to abort, since nodes inside it will first transition to Failing in one round, then their aborts will be called in the next? or maybe that was just an artifact for not having recursive abort? I would rather keep this change in a separate PR though. It seems risky if we get it wrong... and I would like this PR to go in ASAP.

Reply:
ok

Reply:
the only thing is, if we want to keep the state running, then i will need to add execution error to dynamic node state

Reply:
Agreed... let's do that... in the Dynamic-PR